packages/server-wallet/src/evm-validator.ts

```diff
@@ -44,7 +44,18 @@ export const validateTransitionWithEVM = async (
     Uint8Array.from(Buffer.from(bytecode.substr(2), 'hex')),
     Uint8Array.from(Buffer.from(data ? data.toString().substr(2) : '0x00', 'hex'))
   );
-
   // We need to ensure the result is the correct length otherwise we might be interpreting a failed assertion
-  return result.length === 32 && (utils.defaultAbiCoder.decode(['bool'], result)[0] as boolean);
+  const transitionPassed =
+    result.length === 32 && (utils.defaultAbiCoder.decode(['bool'], result)[0] as boolean);
+
+  if (!transitionPassed) {
+    // TODO: Figure out the proper encoding.
+    // Right now the revert reason is readable but slightly garbled
+    logger.error(`Call to ValidTransition failed in the EVM`, {
+      result: new TextDecoder().decode(result),
```

Comment:
```suggestion
      result: new TextDecoder().decode(result).replace(/\0/g, ''),
```

Based on
```
> decoded = new TextDecoder().decode(Buffer.from(input)).replace(
'\b�y�\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000 \u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000(CountingApp: Counter must be incremented\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000'
> decoded.replace(/\0/g, '')
```

Reply:
(It looks like the `result` buffer contains a lot of null characters, perhaps because the first few bytes do not represent valid UTF-8 or something like that?)

Reply:
`\b�y� (CountingApp: Counter must be incremented` is actually what I initially saw when logging out with `console.log`. It appears that  using `pino` causes `\u0000` to be logged out.

Reply:
My guess is that `console.log` prints nothing over and over when it sees the `null` character, but `pino` prints the escaped `\u0000` version of the `null` character.

Reply:
If you ask `pino` to `prettyPrint`, for example, it does not print `\u0000`:
```
> pino = require('pino')({prettyPrint: true});
Pino {
  // snip
}
> pino.info(decoded)
[1603313859603] INFO  (45642 on MacBook-Pro-10.local):�y� (CountingApp: Counter must be incremented
undefined
```

Reply:
I've done something similar in 83cc99e to filter out the null bytes